### PR TITLE
Restructure hot mode test so it runs interactively.

### PR DESCRIPTION
**Pull request by aam**
This allows to add a benchmark for hot reload after actual source code change.

### CODE REVIEW

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 1]
```diff
@@ -162,12 +167,17 @@ class HotRunner extends ResidentRunner {
       printStatus('Benchmarking hot reload');
       // Measure time to perform a hot reload.
       await restart(fullRestart: false);
-      printStatus('Benchmark completed. Exiting application.');
-      await _cleanupDevFS();
-      await stopEchoingDeviceLog();
-      await stopApp();
+      if (stayResident)
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/run_hot.dart]: nit: when you have to use `{}` in `else`, also use `{}` for `if`, even if it's single-line. That's our style guide, sorry ¯\_(ツ)_/¯

File: dev/devicelab/lib/tasks/hot_mode_tests.dart [hunk 2]
```diff
@@ -10,30 +12,96 @@ import '../framework/adb.dart';
 import '../framework/framework.dart';
 import '../framework/utils.dart';
 
+final Directory _editedFlutterGalleryDir = dir(path.join(Directory.systemTemp.path, 'edited_flutter_gallery'));
+final Directory flutterGalleryDir = dir(path.join(flutterDirectory.path, 'examples/flutter_gallery'));
+
+void _copy(Directory source, Directory target) {
```
Comment (yjbanov) [inline dev/devicelab/lib/tasks/hot_mode_tests.dart]: Could you please move this into `lib/framework/utils.dart`? There's already a `copy` function, so many rename this to `recursiveCopy`?

**Review by yjbanov — APPROVED** [review]
Do we expect any movement in the benchmark numbers?

LGTM

**Comment by aam** [comment]
> Do we expect any movement in the benchmark numbers?

No, we don't.

File: dev/devicelab/lib/tasks/hot_mode_tests.dart (hunk 2 above)
Comment (aam) [reply dev/devicelab/lib/tasks/hot_mode_tests.dart]: Done

File: packages/flutter_tools/lib/src/run_hot.dart (hunk 1 above)
Comment (aam) [reply packages/flutter_tools/lib/src/run_hot.dart]: Done

**Pull request merged**